gc.c

```diff
@@ -9316,6 +9315,8 @@ gc_start(rb_objspace_t *objspace, unsigned int reason)
 
     gc_exit(objspace, gc_enter_event_start, &lock_lev);
     return TRUE;
+
+    cr->mfd = prev_mfd;
```

Comment:
This should be **before** `return TRUE`.

Reply:
🤦‍♂️